raiden-ts/tests/unit/raiden.spec.ts

```diff
@@ -120,9 +123,102 @@ function makeDummyDependencies(): RaidenEpicDeps {
 }
 
 describe('Raiden', () => {
+  function initEpicMock(action$: Observable<RaidenAction>): Observable<raidenSynced> {
+    return action$.pipe(
+      filter(raidenStarted.is),
+      mapTo(
+        raidenSynced({
+          tookMs: 9,
+          initialBlock: 1,
+          currentBlock: 10,
+        }),
+      ),
+    );
+  }
+  const token = makeAddress();
+  const tokenNetwork = makeAddress();
+
   test('address', () => {
     const deps = makeDummyDependencies();
     const raiden = new Raiden(dummyState, deps, combineRaidenEpics(of(dummyEpic)), dummyReducer);
     expect(raiden.address).toBe(dummyState.address);
   });
+
+  test('start', async () => {
+    const deps = makeDummyDependencies();
+    const raiden = new Raiden(
+      dummyState,
+      deps,
+      combineRaidenEpics(of(initEpicMock)),
+      dummyReducer,
+    );
+    expect(raiden.network).toEqual({ name: 'test', chainId: 1337 });
+    expect(raiden.log).toEqual(expect.objectContaining({ name: `raiden:${raiden.address}` }));
+    expect(raiden.started).toBeUndefined();
+    await expect(raiden.start()).resolves.toBeUndefined();
+    expect(raiden.started).toEqual(true);
+  });
+
+  test('stop', async () => {
+    const deps = makeDummyDependencies();
+
+    const raiden = new Raiden(
+      dummyState,
+      deps,
+      combineRaidenEpics(of(initEpicMock)),
+      dummyReducer,
+    );
+    await expect(raiden.start()).resolves.toBeUndefined();
+    expect(raiden.started).toEqual(true);
+    const lastPromise = raiden.action$.toPromise();
+    await expect(raiden.stop()).resolves.toBeUndefined();
+    await expect(lastPromise).resolves.toEqual(raidenShutdown({ reason: ShutdownReason.STOP }));
+    expect(raiden.started).toBe(false);
+  });
+
+  test('updateConfig', async () => {
+    const deps = makeDummyDependencies();
+    const raiden = new Raiden(
+      dummyState,
+      deps,
+      combineRaidenEpics(of(initEpicMock)),
+      dummyReducer,
+    );
+    await expect(raiden.start()).resolves.toBeUndefined();
+    const configPromise = raiden.action$.pipe(first(raidenConfigUpdate.is)).toPromise();
+    expect(raiden.started).toEqual(true);
+    const mediationFees = { [token]: { flat: 400 } };
+    raiden.updateConfig({ mediationFees });
+    await expect(configPromise).resolves.toEqual(
+      raidenConfigUpdate(
+        expect.objectContaining({
+          mediationFees: { [token]: { flat: 400 } },
+        }),
+      ),
+    );
+  });
+
+  test('monitorToken', async () => {
+    const deps = makeDummyDependencies();
+    deps.registryContract = {
+      token_to_token_networks: jest.fn().mockImplementation(async () => tokenNetwork),
```

Comment:
```suggestion
      token_to_token_networks: jest.fn(async () => tokenNetwork),
```

Reply:
Could we let it remain like this I could change it in subsequent commits ?